packages/smart-wallet/src/smartWallet.js

```diff
@@ -140,10 +150,47 @@ const { Fail, quote: q } = assert;
  *   liveOfferSeats: WeakMapStore<import('./offers.js').OfferId, UserSeat<unknown>>,
  * }>} ImmutableState
  *
+ * @typedef {BrandDescriptor & { purse: Purse }} PurseRecord
  * @typedef {{
  * }} MutableState
  */
 
+/**
+ * NameHub reverse-lookup, finding 0 or more names for a target value
+ *
+ * XXX move to nameHub.js?
+ *
+ * @param {unknown} target - passable Key
+ * @param {ERef<NameHub>} nameHub
+ */
+const namesOf = async (target, nameHub) => {
+  const entries = await E(nameHub).entries();
+  const matches = [];
+  for (const [name, candidate] of entries) {
+    if (candidate === target) {
+      matches.push(name);
+    }
+  }
+  return harden(matches);
+};
+
+/**
+ * Check that an issuer and its brand belong to each other.
+ *
+ * XXX move to ERTP?
+ *
+ * @param {Issuer} issuer
+ * @param {Brand} brand
+ */
+const mutualCheck = async (issuer, brand) => {
```

Comment:
"assert" would make clear that this does nothing but throw. and the verb should come first.
```suggestion
const assertMutual = async (issuer, brand) => {
```